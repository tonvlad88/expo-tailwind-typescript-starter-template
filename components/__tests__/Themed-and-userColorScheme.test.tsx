import React from "react";
import renderer, { act } from "react-test-renderer";
import { Text, View } from "../Themed";
import { useColorScheme as useColorSchemeHook } from "../useColorScheme";

jest.mock("../useColorScheme");
const mockedUseColorScheme = useColorSchemeHook as jest.Mock;

describe("Themed components & hooks", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("returns lightColor when provided and theme is light", () => {
    mockedUseColorScheme.mockReturnValue("light");
    expect(Text({ lightColor: "orange" }).props.style).toBeDefined();
  });

  it("returns darkColor when provided and theme is dark", () => {
    mockedUseColorScheme.mockReturnValue("dark");
    expect(Text({ darkColor: "blue" }).props.style).toBeDefined();
  });

  it("falls back to Colors.light when no prop is provided", () => {
    mockedUseColorScheme.mockReturnValue("light");
    expect(Text({}).props.style).toBeDefined();
  });

  it("falls back to Colors.dark when no prop is provided", () => {
    mockedUseColorScheme.mockReturnValue("dark");
    expect(Text({}).props.style).toBeDefined();
  });

  it("renders Text with provided lightColor", () => {
    mockedUseColorScheme.mockReturnValue("light");
    let tree;
    act(() => {
      tree = renderer.create(<Text lightColor="orange">Hello</Text>);
    });
    expect(tree!.toJSON()).toMatchSnapshot();
  });

  it("renders Text with fallback color", () => {
    mockedUseColorScheme.mockReturnValue("light");
    let tree;
    act(() => {
      tree = renderer.create(<Text>Hello</Text>);
    });
    expect(tree!.toJSON()).toMatchSnapshot();
  });

  it("renders View with provided lightColor", () => {
    mockedUseColorScheme.mockReturnValue("light");
    let tree;
    act(() => {
      tree = renderer.create(<View lightColor="pink" style={{ padding: 5 }} />);
    });
    expect(tree!.toJSON()).toMatchSnapshot();
  });

  it("renders View with fallback backgroundColor", () => {
    mockedUseColorScheme.mockReturnValue("dark");
    let tree;
    act(() => {
      tree = renderer.create(<View />);
    });
    expect(tree!.toJSON()).toMatchSnapshot();
  });

  // ✅ NEW TEST — covers the ?? 'light' fallback branch
  it("renders with 'light' fallback when useColorScheme returns undefined", () => {
    mockedUseColorScheme.mockReturnValue(undefined);
    let tree;
    act(() => {
      tree = renderer.create(<Text>Hello</Text>);
    });
    expect(tree!.toJSON()).toMatchSnapshot();
  });

  it("useColorScheme hook returns light", () => {
    mockedUseColorScheme.mockReturnValue("light");
    expect(mockedUseColorScheme()).toBe("light");
  });

  it("useColorScheme hook returns dark", () => {
    mockedUseColorScheme.mockReturnValue("dark");
    expect(mockedUseColorScheme()).toBe("dark");
  });

  it("useColorScheme hook returns undefined when system is undefined", () => {
    mockedUseColorScheme.mockReturnValue(undefined);
    expect(mockedUseColorScheme()).toBeUndefined();
  });
});
