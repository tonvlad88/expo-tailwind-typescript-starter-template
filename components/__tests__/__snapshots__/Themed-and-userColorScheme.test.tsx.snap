// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Themed components & hooks renders Text with fallback color 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Hello
</Text>
`;

exports[`Themed components & hooks renders Text with provided lightColor 1`] = `
<Text
  style={
    [
      {
        "color": "orange",
      },
      undefined,
    ]
  }
>
  Hello
</Text>
`;

exports[`Themed components & hooks renders View with fallback backgroundColor 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#000",
      },
      undefined,
    ]
  }
/>
`;

exports[`Themed components & hooks renders View with provided lightColor 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "pink",
      },
      {
        "padding": 5,
      },
    ]
  }
/>
`;

exports[`Themed components & hooks renders with 'light' fallback when useColorScheme returns undefined 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      undefined,
    ]
  }
>
  Hello
</Text>
`;
